service-impl/audit-service-impl/stroom-audit-service/src/main/java/uk.gov.gchq.palisade.audit.service.impl/StroomSimpleAuditService.java

```diff
@@ -0,0 +1,249 @@
+        });
+        //handler for ReadRequestReceivedAuditRequest
+        DISPATCH.put(ReadRequestReceivedAuditRequest.class, (o) -> {
+            requireNonNull(o, "ReadRequestReceivedAuditRequest");
+            ReadRequestReceivedAuditRequest readRequestReceivedAuditRequest = (ReadRequestReceivedAuditRequest) o;
+            final String msg = " 'readRequestReceived' "
+                    + readRequestReceivedAuditRequest.getId() + " "
+                    + readRequestReceivedAuditRequest.getRequestId() + " "
+                    + readRequestReceivedAuditRequest.getOriginalRequestId() + " "
+                    + readRequestReceivedAuditRequest.getResource().toString() + " ";
+            logToStroom(ReadRequestReceivedAuditRequest.class, msg, readRequestReceivedAuditRequest.getRequestId(), null);
+        });
+        //handler for ReadResponseAuditRequest
+        DISPATCH.put(ReadResponseAuditRequest.class, (o) -> {
+            requireNonNull(o, "ReadResponseAuditRequest");
+            ReadResponseAuditRequest readResponseAuditRequest = (ReadResponseAuditRequest) o;
+            final String msg = " 'readResponseAuditRequest' "
+                    + readResponseAuditRequest.getId() + " "
+                    + readResponseAuditRequest.getRequestId() + " "
+                    + readResponseAuditRequest.getOriginalRequestId() + " "
+                    + readResponseAuditRequest.getResource().toString() + " ";
+            logToStroom(ReadResponseAuditRequest.class, msg, readResponseAuditRequest.getRequestId(), null);
+        });
+    }
+
+    public StroomSimpleAuditService() {
+        //Create the EVENT_LOGGING_SYSTEM that is logging the authenticat event
+        EVENT_LOGGING_SYSTEM.setName("Palisade Audit Service");
+        EVENT_LOGGING_SYSTEM.setEnvironment("Test");
+        String hostName = "";
+        String hostAddress = "";
+
+
+    }
+
+    static String auditLogContext(final AuditRequestWithContext auditRequestWithContext) {
+
+        final String msg = " 'userId' " + auditRequestWithContext.getUserId().getId()
+                + " 'purpose' " + auditRequestWithContext.getContext().getPurpose()
+                + "' resourceId '" + auditRequestWithContext.getResourceId()
+                + "' id '" + auditRequestWithContext.getId()
+                + "' originalRequestId '" + auditRequestWithContext.getOriginalRequestId();
+        return msg;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> audit(final AuditRequest request) {
+        requireNonNull(request, "The audit request can not be null.");
+        Consumer<Object> handler = DISPATCH.get((request.getClass()));
+        if (handler != null) {
+            handler.accept(request);
+        } else {
+            //received an AuditRequest derived class that is not defined as a Handler above.
+            //need to add handler for this class.
+            LOGGER.error("handler == null for " + request.getClass().getName());
+
+        }
```

Comment:
Maybe there could be a default dispatcher if no handler is registered?

Reply:
Looking at the DISPATCH map, there is quite a bit of repeated code around audit requests that are cast to AuditRequestWithContext. Suggest a refactor to remove these common cases and see if this can be made the default.